api/rest/execInit.go

```diff
@@ -46,13 +47,16 @@ func HandleInit(c *gin.Context) {
 		return
 	}
 
+	kubeConfigParams := initConfigParams.KubeConfigParams
+	kubeConfigParams.BearerToken = token
+
 	execRequest := handleContainerResolve(c, token, initConfigParams.ContainerName)
 	if execRequest == nil {
 		rest.WriteResponse(c, http.StatusInternalServerError, "Could not retrieve exec request")
 		return
 	}
 
-	err := HandleKubeConfigCreation(&initConfigParams.KubeConfigParams, token, execRequest.ContainerName)
+	err := HandleKubeConfigCreation(&kubeConfigParams, execRequest)
```

Comment:
^ that's not tested, maybe I missed some point and it's better to avoid extending. If you have any concern about it, feel free to just ignore the proposal

Reply:
Mostly just me avoiding making too many changes for a simple fix :) 

I did find it strange we had two almost identical structs.